### PR TITLE
docs: Add references instead of links in theory reference pages.

### PR DESCRIPTION
**Pull request by aniemetz**

### CODE REVIEW

File: docs/references.bib [hunk 1]
```diff
@@ -6,3 +6,101 @@ @article{IEEE754
...
+  publisher = {Springer},
+  year      = {2017},
+  url       = {https://doi.org/10.1007/978-3-319-63046-5\_10},
+  doi       = {10.1007/978-3-319-63046-5\_10},
+  timestamp = {Wed, 25 Sep 2019 18:19:14 +0200},
+  biburl    = {https://dblp.org/rec/conf/cade/MengRTB17.bib},
+  bibsource = {dblp computer science bibliography, https://dblp.org}
+}
+
+@inproceedings{ReynoldsIS016,
```
Comment (nafur) [inline docs/references.bib:49]: Any reason why this particular reference is missing the fourth author in the name?

File: docs/references.bib [hunk 2]
```diff
@@ -6,3 +6,101 @@ @article{IEEE754
...
+  author    = {Kshitij Bansal and
+               Clark W. Barrett and
+               Andrew Reynolds and
+               Cesare Tinelli},
+  title     = {A New Decision Procedure for Finite Sets and Cardinality Constraints
+               in {SMT}},
+  journal   = {CoRR},
+  volume    = {abs/1702.06259},
+  year      = {2017},
+  url       = {http://arxiv.org/abs/1702.06259},
```
Comment (nafur) [inline docs/references.bib:20]:
```suggestion
```

File: docs/references.bib [hunk 3]
```diff
@@ -6,3 +6,101 @@ @article{IEEE754
...
+
+@inproceedings{MengRTB17,
+  author    = {Baoluo Meng and
+               Andrew Reynolds and
+               Cesare Tinelli and
+               Clark W. Barrett},
+  editor    = {Leonardo de Moura},
+  title     = {Relational Constraint Solving in {SMT}},
+  booktitle = {Automated Deduction - {CADE} 26 - 26th International Conference on
+               Automated Deduction, Gothenburg, Sweden, August 6-11, 2017, Proceedings},
```
Comment (nafur) [inline docs/references.bib:36]:
```suggestion
  booktitle = {Automated Deduction - {CADE} 26 - 26th International Conference on
               Automated Deduction},
```

File: docs/references.bib [hunk 4]
```diff
@@ -6,3 +6,101 @@ @article{IEEE754
...
+  series    = {Lecture Notes in Computer Science},
+  volume    = {10395},
+  pages     = {148--165},
+  publisher = {Springer},
+  year      = {2017},
+  url       = {https://doi.org/10.1007/978-3-319-63046-5\_10},
+  doi       = {10.1007/978-3-319-63046-5\_10},
```
Comment (nafur) [inline docs/references.bib:43]:
```suggestion
  doi       = {10.1007/978-3-319-63046-5_10},
```

File: docs/references.bib [hunk 5]
```diff
@@ -6,3 +6,101 @@ @article{IEEE754
...
+  author    = {Andrew Reynolds and
+               Radu Iosif and
+               Cristina Serban and
+               Tim King},
+  editor    = {Cyrille Artho and
+               Axel Legay and
+               Doron Peled},
+  title     = {A Decision Procedure for Separation Logic in {SMT}},
+  booktitle = {Automated Technology for Verification and Analysis - 14th International
+               Symposium, {ATVA} 2016, Chiba, Japan, October 17-20, 2016, Proceedings},
```
Comment (nafur) [inline docs/references.bib:59]:
```suggestion
  booktitle = {Automated Technology for Verification and Analysis - 14th International
               Symposium, {ATVA} 2016},
```

File: docs/references.bib [hunk 6]
```diff
@@ -6,3 +6,101 @@ @article{IEEE754
...
+  series    = {Lecture Notes in Computer Science},
+  volume    = {9938},
+  pages     = {244--261},
+  year      = {2016},
+  url       = {https://doi.org/10.1007/978-3-319-46520-3\_16},
+  doi       = {10.1007/978-3-319-46520-3\_16},
```
Comment (nafur) [inline docs/references.bib:65]:
```suggestion
  doi       = {10.1007/978-3-319-46520-3_16},
```

File: docs/references.bib [hunk 7]
```diff
@@ -6,3 +6,101 @@ @article{IEEE754
...
+}
+
+@inproceedings{ReynoldsB15,
+  author    = {Andrew Reynolds and
+               Jasmin Christian Blanchette},
+  editor    = {Amy P. Felty and
+               Aart Middeldorp},
+  title     = {A Decision Procedure for (Co)datatypes in {SMT} Solvers},
+  booktitle = {Automated Deduction - {CADE-25} - 25th International Conference on
+               Automated Deduction, Berlin, Germany, August 1-7, 2015, Proceedings},
```
Comment (nafur) [inline docs/references.bib:78]:
```suggestion
  booktitle = {Automated Deduction - {CADE-25} - 25th International Conference on
               Automated Deduction},
```

File: docs/references.bib [hunk 8]
```diff
@@ -6,3 +6,101 @@ @article{IEEE754
...
+  series    = {Lecture Notes in Computer Science},
+  volume    = {9195},
+  pages     = {197--213},
+  publisher = {Springer},
+  year      = {2015},
+  url       = {https://doi.org/10.1007/978-3-319-21401-6\_13},
+  doi       = {10.1007/978-3-319-21401-6\_13},
```
Comment (nafur) [inline docs/references.bib:85]:
```suggestion
  doi       = {10.1007/978-3-319-21401-6_13},
```

File: docs/references.bib [hunk 9]
```diff
@@ -6,3 +6,101 @@ @article{IEEE754
...
+  author    = {Clark W. Barrett and
+               Igor Shikanian and
+               Cesare Tinelli},
+  title     = {An Abstract Decision Procedure for a Theory of Inductive Data Types},
+  journal   = {J. Satisf. Boolean Model. Comput.},
+  volume    = {3},
+  number    = {1-2},
+  pages     = {21--46},
+  year      = {2007},
+  url       = {https://doi.org/10.3233/sat190028},
```
Comment (nafur) [inline docs/references.bib:101]:
```suggestion
```

**Review by nafur — CHANGES_REQUESTED** [review]
I realize that some of the proposed changes to the bibtex are personal preference. The dois with underscores are simply broken, though.

**Review by aniemetz — COMMENTED** [review]
@nafur I copied the bibtex from dblp. Except for the fixes in the DOIs, is there a reason why you suggested your changes in the references? For consistence? I usually just keep them as is from dblp (except for removing the URL, that is sometimes reasonable when the DOI and the URL are the same and show up twice in the generated reference).

**Comment by nafur** [comment]
> @nafur I copied the bibtex from dblp. Except for the fixes in the DOIs, is there a reason why you suggested your changes in the references? For consistence? I usually just keep them as is from dblp (except for removing the URL, that is sometimes reasonable when the DOI and the URL are the same and show up twice in the generated reference).

This was/is just personal preference. I don't like having the same link multiple times (url + doi, url + arxiv, etc), and I don't like these overly long booktitles that include the date and place of the conference. But being consistent with dblp makes sense to me as well.

**Review by nafur — APPROVED** [review]
LGTM.

**Comment by aniemetz** [comment]
> > @nafur I copied the bibtex from dblp. Except for the fixes in the DOIs, is there a reason why you suggested your changes in the references? For consistence? I usually just keep them as is from dblp (except for removing the URL, that is sometimes reasonable when the DOI and the URL are the same and show up twice in the generated reference).
> 
> This was/is just personal preference. I don't like having the same link multiple times (url + doi, url + arxiv, etc), and I don't like these overly long booktitles that include the date and place of the conference. But being consistent with dblp makes sense to me as well.

OK, let's remove the duplicate links but keep the rest as in dblp.

File: docs/references.bib (hunk 1 above)
Comment (aniemetz) [reply docs/references.bib:48]: No idea. That's how it is in dblp. I can change it.

**Pull request merged**